### ISSUE TITLE
Style/ConditionalAssignment auto correct removes nested unless inside else

### ISSUE THREAD

**Issue by DMA57361** [rubocop-hq/rubocop#7290]
We've found some code that the Style/ConditionalAssignment auto-correction ends up breaking by removing a nested unless branch from within the else.

In our case we've changed the nested unless to be another elsif and all is well, but thought I'd report the breaking auto-correct.

--------

## Expected / Actual behavior

A nested unless inside an else is removed by the Style/ConditionalAssignment cop, resulting in a change of behaviour.

## Steps to reproduce the problem

Given a file called `example.rb` with this content:

``` ruby
def do_some_work(var)
  if var.any?(:prob_a_check)
    @errors << 'Problem A'
  elsif var.any?(:prob_a_check)
    @errors << 'Problem B'
  else
    unless var.all?(:save)
      @errors << 'Save failed'
    end
  end
end
```

Run `rubocop example.rb --auto-correct --only Style/ConditionalAssignment`

```
$ rubocop example.rb --auto-correct --only Style/ConditionalAssignment
Inspecting 1 file
C

Offenses:

example.rb:2:3: C: [Corrected] Style/ConditionalAssignment: Use the return of the conditional for variable assignment and comparison.
  if var.any?(:prob_a_check) ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
$
```

and the example.rb file is left in this state:
``` ruby
def do_some_work(var)
  @errors << if var.any?(:prob_a_check)
    'Problem A'
  elsif var.any?(:prob_b_check)
    'Problem B'
  else
    @errors << 'Save failed'
             end
end
```

Note that the behaviour of the else branch has changed. The unless has been removed and the `@errors <<` removed from the other branches still remains within the else.

## RuboCop version

```
$ bundle exec rubocop -V
0.74.0 (using Parser 2.6.3.0, running on ruby 2.6.3 x86_64-darwin18)
```
